web/styles/inputs.css

```diff
@@ -0,0 +1,86 @@
+.input-element-wrapper {
+    display: grid;
+    grid-template: "icon input button" auto / auto minmax(0, 1fr) auto;
+    grid-template-rows: 1fr;
+    align-items: center;
+}
+
+.input-icon {
+    pointer-events: none;
+    grid-area: icon;
+    margin-left: 0.5em;
+    color: var(--color-text-input);
+    z-index: 1;
+}
+
+.input-active-styles {
+    color: var(--color-text-input-focus);
+    background-color: var(--color-background-input-focus);
+    border-color: var(--color-border-input-focus);
+}
+
+.input-element-wrapper .input-element {
+    grid-column: icon-start / button-end;
```

Comment:
I think you can avoid `grid-row: 1` by changing `grid-column` to `grid-area` here.

Reply:
Done, with the help of the new template column definition from: https://github.com/zulip/zulip/pull/34656#discussion_r2146442148